Source/SA/RFmx Instr/RFmxInstr.cs

```diff
@@ -0,0 +1,112 @@
+        public struct InstrumentConfiguration
+        {
+            public LocalOscillatorSharingMode LOSharingMode;
+            public LocalOscillatorRoutingConfiguration[] LORoutingConfigurations;
+            public LocalOscillatorFrequencyOffsetConfiguration LOOffsetConfiguration;
+
+            public static InstrumentConfiguration GetDefault()
+            {
+                return new InstrumentConfiguration
+                {
+                    LOSharingMode = LocalOscillatorSharingMode.Automatic,
+                    LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[] { LocalOscillatorRoutingConfiguration.GetDefault() },
+                    LOOffsetConfiguration = LocalOscillatorFrequencyOffsetConfiguration.GetDefault()
+                };
+            }
+
+            public static InstrumentConfiguration GetDefault(RFmxInstrMX sessionHandle)
+            {
+                InstrumentConfiguration instrConfig = GetDefault(); // covers case for sub6 instruments with a single configurable LO
+                // lo routing configurations will now be overridden if the instrument has number of LOs != 1
+                sessionHandle.GetInstrumentModel("", out string instrumentModel);
+                if (instrumentModel.Equals("NI PXIe-5830"))
+                    instrConfig.LORoutingConfigurations[0].ChannelName = "LO2";
+                else if (Regex.IsMatch(instrumentModel, "NI PXIe-5831"))
+                {
+                    LocalOscillatorRoutingConfiguration lo1RoutingConfig = LocalOscillatorRoutingConfiguration.GetDefault();
+                    lo1RoutingConfig.ChannelName = "LO1";
+                    LocalOscillatorRoutingConfiguration lo2RoutingConfig = LocalOscillatorRoutingConfiguration.GetDefault();
+                    lo2RoutingConfig.ChannelName = "LO2";
+                    instrConfig.LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[] { lo1RoutingConfig, lo2RoutingConfig };
+                }
+                else if (Regex.IsMatch(instrumentModel, "NI PXIe-5(82.|645R)")) // matches on any baseband instrument without an LO
+                {
+                    instrConfig.LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[0]; // baseband instruments have no LOs
+                }
+                return instrConfig;
+            }
+        }
+        #endregion
+
+        #region Instrument Configurations
+        public static void ConfigureInstrument(RFmxInstrMX instrHandle, InstrumentConfiguration instrConfig)
+        {
+            instrHandle.GetInstrumentModel("", out string instrumentModel);
+
+            if (Regex.IsMatch(instrumentModel, "NI PXIe-5(82.|645R)")) // matches on any baseband instrument without an LO
+                return; // return early since the instrument doesn't have any LOs that can be configured
+
+            // Set channel agnostic auto SG SA shared property
+            if (Regex.IsMatch(instrumentModel, "NI PXIe-58[34]."))

```

Comment:
Shouldn't there be a bracket?
```suggestion
            if (Regex.IsMatch(instrumentModel, "NI PXIe-58[34]."))
            {
```

Reply:
the switch statement falls into the if, but i will add a bracket so its clear

Reply:
Hmm, I guess I am thinking from C++ days. It was possible to do this for a one-liner but I don't think it was possible for multiline code. I didn't realize this worked in C#; either way, I think that I still prefer the precedent for explicit brackets except on one-liners.